library(shiny)
library(shinythemes)
library(ggplot2)
library(RColorBrewer)
library(leaflet)


# input your data file
dataset <- read.csv("completedata.txt", header= TRUE, sep=",")
geocode <- read.table("geoMerge.txt", header=TRUE, sep="\t")
geocode <- na.omit(geocode)

#make for drop down menu to be in alphabetical order
dataset <- with(dataset,  dataset[order(Common.Name) , ])

## Define UI for application that plots random distributions

shinyUI(fluidPage(theme = shinytheme("cerulean"),
                  #Appication title
                  titlePanel("SRAexplorer!"),
                
 
  
  sidebarPanel (
    
    # sliderInput('sampleSize','Sample Size', min=1, max=nrow(dataset),
    #             value=min(1000,nrow(dataset)),
    #             step=500,
    #             round=0),
    
    # here is the information that will go in the pull down menu and you are creating a variable "x" 
    # which will be used for doing the filtering in the server side.
    h4(strong("Timeline Options")),
    radioButtons('rb', 'Select a year:',unique(as.character(dataset$Date))),
    br(),
    h4(strong("Map Options")),
    sliderInput("freqslide", "Frequency range of entries:", min=min(geocode$freq), max=max(geocode$freq), value=range(geocode$freq), step=1000),
    selectInput("colours", "Select a Colour Scheme:",
                rownames(subset(brewer.pal.info, category %in% c("seq", "div")))),
    checkboxInput("legend", "Show legend?", TRUE),
    br(),
    h4(strong("Organism Specific Options")),
    selectInput('x','Select an organism:',unique(as.character(dataset$Common.Name))),
    selectInput('y','Select a Platform:',c('All',unique(as.character(dataset$Model))))
   

    
  ),
  
  ## Show a plot of the generated distribution
  
  mainPanel(
   tabsetPanel(
    tabPanel("Background", h3("What is the SRAexplorer?", align = "left"),
                      img(src='dna.png',height=200, width=200, align = "left"),
                      br(),
                      p("The SRAexplorer is an application designed to", strong("visualise and explore"), "the data uploaded to NCBI's Sequence Read Archive (SRA). 
                        The SRA is a bioinformatics database that provides a public repository for next generation sequencing data, especially the reads
                        generated by high-throughput sequencing. The metadata on the SRA has yet to be entirely catalogued, so categorising the database would allow for a clearer picture
                        of the data available."),
                      br(),
                      br(),
                      h3("Features include..."),
                      p("*Interactive map to display which institutes upload to the SRA"),
                      p("*A timeline of the number of bases submitted by platform"),
                      p("*The most popular next generation sequencing platforms"),
                      p("*Organism specific metadata graphs")
                      
                      ),
             tabPanel("Timeline",
                      br(),
                      p("The SRA database was designed in early 2007 and was launched by NCBI for user submissions."),
                      br(),
                      h5("Number of Bases Sequenced by Each Platform between 2007:2017"),
                      plotOutput("base"), 
                      br(), 
                      h5("Number of bases each platform model sequenced by year selected"),
                      plotOutput("modelGraph"),
                      br(),
                      h5("Top 10 Organisms for selected year"),
                      plotOutput("orgGraph")),
             
             
             tabPanel("Interactive Map", 
                      br(),
                      p("Institutes that have submitted to the SRA are highlighted in the map below"),
                      br(),
                      h5("World Map"),
                      leafletOutput("mymap"),
                      br(),
                      h5("Table of top 30 submitters to the SRA"),
                      tableOutput("mapTable")),
                      
             tabPanel("Organism Specific Data",
                      br(),
                      p("Explore organism specific metadata."),
                      br(),
                      h5("Word Cloud for the current top 75 of Organisms (excluding human)"),
                      plotOutput('cloud'),
                      h5("Number of Bases Sequenced by each Platform based on Organism"),
                      plotOutput('companyPlot',height="500px"),
                      h5("Number of entries for specific organism by library source"),
                      plotOutput('libSourcePlot', height="500px"),
                      h5("Number of entries per year by specific organism"),
                      plotOutput("orgYearPlot")
                      
    # I used the line below to check whether the information was being passed and filtered - debug code -
    #textOutput("text1")
 
    
    
    
    )
  
   )
))

)
